swift-ticketing/test/swift_ticketing/worker_test.clj

```diff
@@ -0,0 +1,176 @@
+          (with-redefs
+           [db-ticket/lock-reserved-tickets (constantly selected-tickets)
+            db-ticket/confirm-tickets
+            (fn [_ ticket-ids]
+              (reset! confirm-tickets-args {:ticket-ids ticket-ids}))
+            db-booking/update-booking-status
+            (fn [_ bid status]
+              (reset! update-booking-args {:booking-status status
+                                           :booking-id bid}))]
+
+            (worker/add-book-ticket-request-to-queue
+             message-queue
+             {:booking-id booking-id})
+            (worker/process-ticket-requests
+             worker-id message-queue db-spec nil exit-ch)
+            (Thread/sleep 2000)
+            (is (= {:ticket-ids
+                    (map :ticket-id selected-tickets)}
+                   @confirm-tickets-args))
+            (is (= {:booking-status db-booking/CONFIRMED
+                    :booking-id booking-id}
+                   @update-booking-args)))))
+
+      (testing "Reserve ticket request"
+        (let [booking-id (random-uuid)
+              selected-tickets (map (constantly (factory/mk-ticket))
+                                    (range (inc (rand-int 20))))
+              reserve-tickets-args (atom {})
+              update-booking-args (atom {})]
+
+          (testing "with zero ticket quantity"
+            (with-redefs
+             [db-booking/update-booking-status
+              (fn [_ bid status]
+                (reset! update-booking-args {:booking-status status
+                                             :booking-id bid}))]
+
+              (worker/add-reserve-ticket-request-to-queue
+               message-queue
+               (factory/worker-reserve-ticket-request booking-id []))
+              (worker/process-ticket-requests 1 message-queue db-spec nil exit-ch)
+              (Thread/sleep 2000)
+              (is (= {:booking-status db-booking/REJECTED
+                      :booking-id booking-id}
+                     @update-booking-args))))
+
+          (reset! update-booking-args {})
+
+          (testing "with valid ticket ids"
+            (with-redefs
+             [db-ticket/lock-unbooked-tickets (constantly selected-tickets)
+              db-ticket/reserve-tickets
+              (fn [_ ticket-ids bid _]
+                (reset! reserve-tickets-args {:ticket-ids ticket-ids
+                                              :booking-id bid}))
+              db-booking/update-booking-status
+              (fn [_ bid status]
+                (reset! update-booking-args {:booking-status status
+                                             :booking-id bid}))]
+
+              (worker/add-reserve-ticket-request-to-queue
+               message-queue
+               (factory/worker-reserve-ticket-request
+                booking-id selected-tickets))
+              (worker/process-ticket-requests
+               worker-id message-queue db-spec nil exit-ch)
+              (Thread/sleep 2000)
```

Comment:
This violates the `wait, don't sleep` principle. Change the test to wait on a promise or something to make it less fragile: https://harith-sankalpa.medium.com/wait-do-not-sleep-f916e501878d